te_chef.py

```diff
@@ -35,50 +62,228 @@ def construct_channel(self, **kwargs):
-    for child_source_node in data_source:
+    doc = get_parsed_html_from_url(country_url)
+    categories = doc.select(".sub_cat_listing a")
 
-        if child_source_node.get('children'):
-            child_node = nodes.TopicNode(
-                source_id=child_source_node["id"],
-                title=child_source_node["title"],
-            )
-            parent.add_child(child_node)
-            _build_tree(child_node, child_source_node.get("children", []))
+    for category in categories:
+        url = category["href"]
+        title = category.text.strip()
+        country_node.add_child(scrape_category(title, url))
 
+
+def scrape_category(title, category_url):
+    """
+    title: Culture
+    category_url: http://www.touchableearth.org/china/culture/
+        ... redirects to: http://www.touchableearth.org/china-culture-boys-clothing/
+    """
+    print("  Scraping category node: %s (%s)" % (title, category_url))
+
+    category_node = nodes.TopicNode(source_id=category_url, title=title)
+
+    # Iterate over each item in the "subway" sidebar menu on the left.
+    doc = get_parsed_html_from_url(category_url)
+    content_items = doc.select(".post_title_sub .current_post")
+    slugs_added = set()
+
+    for content in content_items:
+        slug = content.select_one(".get_post_title")["value"]
+
+        # Skip duplicates ... seems like the Touchable Earth website has them!
+        if slug in slugs_added:
+            continue
         else:
-            child_node = nodes.VideoNode(
-                source_id=child_source_node["id"],
-                title=child_source_node["title"],
-                license=child_source_node["license"],
-                description=create_description(child_source_node),
-                derive_thumbnail=True,
-                files=[files.YouTubeVideoFile(youtube_id=child_source_node['youtube_id'])],
-            )
-            for language in child_source_node['subtitle_langs']:
-                child_node.add_file(files.YouTubeSubtitleFile(youtube_id=child_source_node['youtube_id'], language=language))
-            parent.add_child(child_node)
-
-    return parent
+            slugs_added.add(slug)
+
+        title = content.select_one(".get_post_title2")["value"]
+        site_url = content.select_one(".site_url")["value"]
+        url = "%s/%s" % (site_url, slug)
+
+        content_node = scrape_content(title, url)
+        if content_node:
+            category_node.add_child(content_node)
+
+    return category_node
+
+
+def scrape_content(title, content_url):
+    """
+    title: Boys' clothing
+    content_url: http://www.touchableearth.org/china-culture-boys-clothing/
+    """
+    print("    Scraping content node: %s (%s)" % (title, content_url))
+
+    doc = get_parsed_html_from_url(content_url)
+    if not doc:  # 404
+        return None
+
+    description = create_description(doc.select_one(".tab-content"))
+    base_node_attributes = {
+        "source_id": content_url,
```

Comment:
Great suggestion, thanks Jamie, done.

(I had thought of using the slug or not the URL but decided to skip doing that ... the points you mentioned above re: site restructuring and move to HTTPS is a good one.)